server.js

```diff
@@ -177,6 +185,25 @@ app.get('/protected', (req, res) => {
   res.send('Hello! You are viewing protected content.');
 });
 
+const FormData = require('form-data');
+const upload = multer();
+app.use('/api/v0/add', upload.single('file'), async (req, res, next) => {
+  const formData = new FormData();
+
+  formData.append(req.file.fieldname, req.file.buffer, {
+    filename: req.file.fieldname,
+    contentType: 'application/octet-stream',
+  });
+
+  createProxyMiddleware({
+    target: IPFS_URL,
+    onProxyReq: (proxyReq) => {
+      proxyReq.setHeader('Content-Type', formData.getHeaders()['content-type']);
+      formData.pipe(proxyReq);
+    },
+  })(req, res, next);
+});
+
```

Comment:
Though it is good for this PR, for the next one we should avoid local uploads and stream to the backend directly as is.

Something like this.
```suggestion
app.use('/api/v0/add', createProxyMiddleware({
    target: IPFS_URL,
    selfHandleResponse: true
}));
```

Same for `/cat`.